### DISCUSSION TITLE
Password complexity options: haveibeenpwned check

### DISCUSSION THREAD

**Discussion by tyteen4a03** [zitadel/zitadel#8621] [Ideas]
It would be great if Zitadel can automatically check passwords against haveibeenpwned's API, and reject a password if a password shows up in more than x breaches.

**Comment by fforootd** [comment]
This would be indeed a nice feature 👍

**Comment by ruffy91** [comment]
HIBP checks would bring real value to users still relying on passwords. This would also solve the point "8.4. Consider Restriction of Common or Compromised 
Passwords" of your pentest from 2021: https://zitadel.com/blog/[2021-11-11-pentest-results-h1-2021-01.pdf](https://zitadel.com/blog/2021-11-11-pentest-results-h1-2021-01.pdf)
